lxd/storage/drivers/drivers_lvm.go

```diff
@@ -0,0 +1,532 @@
+		}
+		d.logger.Debug("Physical loop file removed", log.Ctx{"file_name": d.config["source"]})
+	}
+
+	// Wipe everything in the storage pool directory.
+	err = wipeDirectory(GetPoolMountPath(d.name))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *lvm) Validate(config map[string]string) error {
+	return nil
+}
+
+// Update updates the storage pool settings.
+func (d *lvm) Update(changedConfig map[string]string) error {
+	if changedConfig["lvm.vg_name"] != "" {
+		_, err := shared.TryRunCommand("vgrename", d.config["lvm.vg_name"], changedConfig["lvm.vg_name"])
+		if err != nil {
+			return errors.Wrapf(err, "Error renaming LVM volume group from %q to %q", d.config["lvm.vg_name"], changedConfig["lvm.vg_name"])
+		}
+		d.logger.Debug("Volume group renamed", log.Ctx{"vg_name": d.config["lvm.vg_name"], "new_vg_name": changedConfig["lvm.vg_name"]})
+	}
+
+	if changedConfig["lvm.thinpool_name"] != "" {
+		_, err := shared.TryRunCommand("lvrename", d.config["lvm.vg_name"], d.config["lvm.thinpool_name"], changedConfig["lvm.thinpool_name"])
+		if err != nil {
+			return errors.Wrapf(err, "Error renaming LVM thin pool from %q to %q", d.config["lvm.thinpool_name"], changedConfig["lvm.thinpool_name"])
+		}
+		d.logger.Debug("Thin pool volume renamed", log.Ctx{"vg_name": d.config["lvm.vg_name"], "thinpool": d.config["lvm.thinpool_name"], "new_thinpool": changedConfig["lvm.thinpool_name"]})
+	}
+
+	return nil
+}
+
+// Mount mounts the storage pool (this does nothing for external LVM pools, but for loopback image
+// LVM pools this creates a loop device).
+func (d *lvm) Mount() (bool, error) {
+	// Open the loop file if the LVM device doesn't exist yet and the source points to a file.
+	if !shared.IsDir(fmt.Sprintf("/dev/%s", d.config["lvm.vg_name"])) && filepath.IsAbs(d.config["source"]) && !shared.IsBlockdevPath(d.config["source"]) {
+		loopFile, err := d.openLoopFile(d.config["source"])
+		if err != nil {
+			return false, err
+		}
+		defer loopFile.Close()
+
+		return true, nil
+	}
+
+	// Activate volume group so that it's device is added to /dev.
+	_, err := shared.TryRunCommand("vgchange", "-ay", d.config["lvm.vg_name"])
+	if err != nil {
+		return false, err
+	}
+
+	return false, nil
+}
+
+// Unmount unmounts the storage pool (this does nothing for external LVM pools, but for loopback
+// image LVM pools this closes the loop device handle if needed).
+func (d *lvm) Unmount() (bool, error) {
+	// If loop backed, force release the loop device.
+	if filepath.IsAbs(d.config["source"]) && !shared.IsBlockdevPath(d.config["source"]) {
+		vgExists, _, _ := d.volumeGroupExists(d.config["lvm.vg_name"])
+		if vgExists {
+			// Deactivate volume group so that it's device is removed from /dev.
+			_, err := shared.TryRunCommand("vgchange", "-an", d.config["lvm.vg_name"])
+			if err != nil {
+				return false, err
+			}
+		}
+
+		err := releaseLoopDev(d.config["source"])
```

Comment:
@stgraber I've left `releaseLoopDev` call in here, but with failure detection, so we should be made aware of any issues with it.